File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Room/AbstractRoom.ts
import {MonsterGroup} from "mods/ModTheSpire/Scripts/Unit/Monster/MonsterGroup";

export enum RoomPhase{
    Combat,
    Event,
    COMPLETE,
    INCOMPLETE
}

export abstract class AbstractRoom{
    //当前房间怪物列表
    public Monsters:MonsterGroup
    //如果要制作怪物房间.请继承该变量为1.
    public Round = 0;
    abstract Phase:RoomPhase;
    abstract Enter();
    abstract End();
    public AddReward(reward:any){}
}

File: Unity/Assets/Scripts/ModTheSpire/Room/AbstractRoom.cs
using ModTheSpire.Card;
using ModTheSpire.Unit.Monster;

namespace ModTheSpire.Room
{
    public abstract class AbstractRoom
    {
        public MonsterGroup Monsters;
        public CardGroup Souls;
        public int Round;
        public abstract RoomPhase Phase { get; }
        public abstract void Enter();
        public abstract void End();

        public virtual void AddReward(object[] reward)
        {
        }
    }
}